### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException: Index 1902 out of bounds for length 1900

### ISSUE THREAD

**Issue by jmatias** [jonathanlermitage/intellij-extra-icons-plugin#119]
Occurs whenever I click "Add".

![image](https://user-images.githubusercontent.com/1302269/221649992-c17b37cb-da28-4764-bc8a-ca7c059ad2cd.png)

## Issues

Please describe the issue:

```stack trace

java.lang.ArrayIndexOutOfBoundsException: Index 1902 out of bounds for length 1900
	at org.apache.batik.ext.awt.image.GraphicsUtil.mult_BYTE_COMP_Data(GraphicsUtil.java:1513)
	at org.apache.batik.ext.awt.image.GraphicsUtil.multiplyAlpha(GraphicsUtil.java:1028)
	at org.apache.batik.ext.awt.image.GraphicsUtil.coerceData(GraphicsUtil.java:1018)
	at org.apache.batik.ext.awt.image.rendered.AffineRed.genRect(AffineRed.java:176)
	at org.apache.batik.ext.awt.image.rendered.AffineRed.copyData(AffineRed.java:99)
```

<details> <summary><b><u>Click here to expand stack trace trace</u></b></summary>
<p>

```stacktrace
Following exceptions happened soon after this one, most probably they are induced.


2023-02-27T13:11:10.359
java.lang.ArrayIndexOutOfBoundsException: Index 1902 out of bounds for length 1900
	at org.apache.batik.ext.awt.image.GraphicsUtil.mult_BYTE_COMP_Data(GraphicsUtil.java:1513)
	at org.apache.batik.ext.awt.image.GraphicsUtil.multiplyAlpha(GraphicsUtil.java:1028)
	at org.apache.batik.ext.awt.image.GraphicsUtil.coerceData(GraphicsUtil.java:1018)
	at org.apache.batik.ext.awt.image.rendered.AffineRed.genRect(AffineRed.java:176)
	at org.apache.batik.ext.awt.image.rendered.AffineRed.copyData(AffineRed.java:99)
	at org.apache.batik.ext.awt.image.rendered.FormatRed.copyData(FormatRed.java:109)
	at org.apache.batik.ext.awt.image.rendered.AbstractRed.getData(AbstractRed.java:533)
	at org.apache.batik.ext.awt.image.GraphicsUtil.drawImage(GraphicsUtil.java:236)
	at org.apache.batik.ext.awt.image.GraphicsUtil.drawImage(GraphicsUtil.java:427)
	at org.apache.batik.gvt.RasterImageNode.primitivePaint(RasterImageNode.java:104)
	at org.apache.batik.gvt.AbstractGraphicsNode.paint(AbstractGraphicsNode.java:529)
	at org.apache.batik.gvt.CompositeGraphicsNode.primitivePaint(CompositeGraphicsNode.java:161)
	at org.apache.batik.gvt.AbstractGraphicsNode.paint(AbstractGraphicsNode.java:529)
	at org.apache.batik.gvt.ImageNode.paint(ImageNode.java:74)
	at org.apache.batik.gvt.CompositeGraphicsNode.primitivePaint(CompositeGraphicsNode.java:161)
	at org.apache.batik.gvt.AbstractGraphicsNode.paint(AbstractGraphicsNode.java:529)
	at org.apache.batik.gvt.CompositeGraphicsNode.primitivePaint(CompositeGraphicsNode.java:161)
	at org.apache.batik.gvt.AbstractGraphicsNode.paint(AbstractGraphicsNode.java:529)
	at org.apache.batik.gvt.CompositeGraphicsNode.primitivePaint(CompositeGraphicsNode.java:161)
	at org.apache.batik.gvt.AbstractGraphicsNode.paint(AbstractGraphicsNode.java:529)
	at org.apache.batik.gvt.CompositeGraphicsNode.primitivePaint(CompositeGraphicsNode.java:161)
	at org.apache.batik.gvt.AbstractGraphicsNode.paint(AbstractGraphicsNode.java:529)
	at org.apache.batik.gvt.CompositeGraphicsNode.primitivePaint(CompositeGraphicsNode.java:161)
	at org.apache.batik.gvt.CanvasGraphicsNode.primitivePaint(CanvasGraphicsNode.java:159)
	at org.apache.batik.gvt.AbstractGraphicsNode.paint(AbstractGraphicsNode.java:529)
	at org.apache.batik.gvt.CompositeGraphicsNode.primitivePaint(CompositeGraphicsNode.java:161)
	at org.apache.batik.gvt.AbstractGraphicsNode.paint(AbstractGraphicsNode.java:529)
	at com.intellij.ui.svg.SvgTranscoderKt.render(SvgTranscoder.kt:281)
	at com.intellij.ui.svg.SvgTranscoderKt.access$render(SvgTranscoder.kt:1)
	at com.intellij.ui.svg.SvgTranscoder$Companion.createImage(SvgTranscoder.kt:100)
	at com.intellij.ui.svg.SvgTranscoder$Companion.createImage$default(SvgTranscoder.kt:75)
	at com.intellij.util.SVGLoaderKt.loadAndCache(SVGLoader.kt:385)
	at com.intellij.util.SVGLoaderKt.access$loadAndCache(SVGLoader.kt:1)
	at com.intellij.util.SVGLoader.loadFromClassResource(SVGLoader.kt:109)
	at com.intellij.util.ImageLoaderKt.loadByDescriptorWithoutCache(ImageLoader.kt:580)
	at com.intellij.util.ImageLoaderKt.loadByDescriptor(ImageLoader.kt:542)
	at com.intellij.util.ImageLoaderKt.access$loadByDescriptor(ImageLoader.kt:1)
	at com.intellij.util.ImageLoader.loadImage(ImageLoader.kt:86)
	at com.intellij.openapi.util.ImageDataByUrlLoader.loadImage(ImageDataByUrlLoader.java:87)
	at com.intellij.openapi.util.CachedImageIcon.loadImage$intellij_platform_util_ui(CachedImageIcon.kt:269)
	at com.intellij.openapi.util.ScaledIconCache.getOrScaleIcon(ScaledIconCache.kt:46)
	at com.intellij.openapi.util.CachedImageIcon.getRealIcon$intellij_platform_util_ui(CachedImageIcon.kt:151)
	at com.intellij.openapi.util.CachedImageIcon.getIconWidth(CachedImageIcon.kt:104)
	at java.desktop/javax.swing.JLabel.setIcon(JLabel.java:414)
	at lermitage.intellij.extra.icons.cfg.dialogs.ModelDialog$ComboBoxRenderer.getListCellRendererComponent(ModelDialog.java:419)
	at java.desktop/javax.swing.plaf.basic.BasicComboBoxUI.getDisplaySize(BasicComboBoxUI.java:1446)
	at java.desktop/javax.swing.plaf.basic.BasicComboBoxUI.getMinimumSize(BasicComboBoxUI.java:999)
	at com.intellij.ide.ui.laf.darcula.ui.DarculaComboBoxUI.getMinimumSize(DarculaComboBoxUI.java:587)
	at java.desktop/javax.swing.JComponent.getMinimumSize(JComponent.java:1801)
	at com.intellij.uiDesigner.core.Util.getMinimumSize(Util.java:13)
	at com.intellij.uiDesigner.core.DimensionInfo.getMinimumSize(DimensionInfo.java:238)
	at com.intellij.uiDesigner.core.HorizontalInfo.getMinimumWidth(HorizontalInfo.java:40)
	at com.intellij.uiDesigner.core.GridLayoutManager.getMinOrPrefSizes(GridLayoutManager.java:640)
	at com.intellij.uiDesigner.core.GridLayoutManager.getPrefSizes(GridLayoutManager.java:627)
	at com.intellij.uiDesigner.core.GridLayoutManager.preferredLayoutSize(GridLayoutManager.java:271)
	at com.intellij.uiDesigner.core.GridLayoutManager.layoutContainer(GridLayoutManager.java:406)
	at java.desktop/java.awt.Container.layout(Container.java:1541)
	at java.desktop/java.awt.Container.doLayout(Container.java:1530)
	at java.desktop/java.awt.Container.validateTree(Container.java:1725)
	at java.desktop/java.awt.Container.validateTree(Container.java:1734)
	at java.desktop/java.awt.Container.validateTree(Container.java:1734)
	at java.desktop/java.awt.Container.validateTree(Container.java:1734)
	at java.desktop/java.awt.Container.validateTree(Container.java:1734)
	at java.desktop/java.awt.Container.validateTree(Container.java:1734)
	at java.desktop/java.awt.Container.validate(Container.java:1660)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2802)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:790)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:731)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:763)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:761)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:760)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:666)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$7(IdeEventQueue.kt:570)
	at com.intellij.openapi.application.impl.ApplicationImpl.withoutImplicitRead(ApplicationImpl.java:1442)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:570)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:68)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:348)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:348)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:343)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:994)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:994)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$4(IdeEventQueue.kt:343)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:385)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:121)
	at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:191)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:236)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:234)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
	at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:234)
	at java.desktop/java.awt.Dialog.show(Dialog.java:1080)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:741)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:466)
	at com.intellij.openapi.ui.DialogWrapper.doShow(DialogWrapper.java:1675)
	at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1633)
	at com.intellij.ide.actions.ShowSettingsUtilImpl.showSettingsDialog(ShowSettingsUtilImpl.java:90)
	at com.intellij.ide.actions.ShowSettingsAction.perform(ShowSettingsAction.java:60)
	at com.intellij.ui.mac.MacOSApplicationProvider$Worker.initMacApplication$lambda$2$lambda$1(MacOSApplicationProvider.kt:71)
	at com.intellij.ui.mac.MacOSApplicationProvider$Worker.submit$lambda$6(MacOSApplicationProvider.kt:170)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:209)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:191)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.openapi.application.impl.ApplicationImpl$3.run(ApplicationImpl.java:456)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:79)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:122)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:41)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:788)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:731)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:758)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:666)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$7(IdeEventQueue.kt:570)
	at com.intellij.openapi.application.impl.ApplicationImpl.withoutImplicitRead(ApplicationImpl.java:1442)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:570)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:68)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:348)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:348)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:343)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:994)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:994)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$4(IdeEventQueue.kt:343)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:385)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

```

</p>
</details>

**plugin version**: 2023.1.1

IntelliJ IDEA 2023.1 EAP (Ultimate Edition)
Build #IU-231.7665.28, built on February 23, 2023
Licensed to IntelliJ IDEA EAP user: Javier Matias
Expiration date: March 25, 2023
Runtime version: 17.0.6+10-b829.1 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 13.2.1
GC: G1 Young Generation, G1 Old Generation
Memory: 8000M
Cores: 10
Metal Rendering is ON
Registry:
    editor.focus.mode.color.light=5a5d63
    ide.editor.tab.selection.animation=true
    debugger.new.tool.window.layout=true
    ide.new.editor.tabs.vertical.borders=true
    ide.experimental.ui=true
    ide.balloon.shadow.size=0
    editor.focus.mode.color.dark=5a5d63
    editor.minimap.enabled=true

Non-Bundled Plugins:
    com.jetbrains.plugins.ini4idea (231.7665.20)
    FrameSwitcher (3.31.0-212)
    Keymap exporter (2.6)
    com.github.copilot (1.2.3.2385)
    org.jetbrains.plugins.go-template (231.7665.25)
    name.kropp.intellij.makefile (231.7665.25)
    org.jetbrains.plugins.rest (231.7665.20)
    org.jetbrains.plugins.go (231.7665.28)
    lermitage.intellij.extra.icons (2023.1.1)
    com.intellij.kubernetes (231.7665.20)
    com.obroom.plugin.toolset (2022.12.1)
    com.intellij.tide (231.7665.20)
    org.intellij.scala (2023.1.9)
    Pythonid (231.7665.28)
    com.intellij.bigdatatools (231.7665.28)
    com.jetbrains.hackathon.indices.viewer (1.23)

Kotlin: 231-1.8.0-release-345-IJ7665.28

:warning: This information is needed in order to determine if this is an IDE bug or a plugin bug, and to find a solution or a workaround.

**Comment by jonathanlermitage** [comment]
Hi! This is a bug with the SVG renderer in IDE 2023 EAP. It cannot render the Caddyfile icon.  
I implemented a workaround in the upcoming release of Extra Icons (planned for 2023/03/11).
I packaged a version here: 
[Extra Icons-2023.1.1.zip](https://github.com/jonathanlermitage/intellij-extra-icons-plugin/files/10842864/Extra.Icons-2023.1.1.zip)

**Comment by jonathanlermitage** [comment]
Can you send me the SVG file? Thx!

**Comment by jmatias** [comment]
![azure-pipelines](https://user-images.githubusercontent.com/1302269/221656056-510665cd-d714-49aa-9267-e497fb85011e.svg)

**Comment by jonathanlermitage** [comment]
Hum, it fails with JetBrains SVG loader AND with the TwelveMonkeys library, in IntelliJ EAP. No problem with IntelliJ 2022.3.2. Something is broken 😭 I will investigate.

**Comment by jonathanlermitage** [comment]
This is weird, the error says `java.lang.NoSuchMethodError: 'org.apache.batik.bridge.BridgeContext com.twelvemonkeys.imageio.plugins.svg.SVGImageReader$Rasterizer.createBridgeContext()'`, but this method exists...

**Comment by jonathanlermitage** [comment]
OK, this is a classloader issue. IntelliJ used to come with some libraries like Apache Batik, Xerces... but with the latest EAP, they changed some things. The libraries are still there, but the ImageIO class (from the Java SDK) no longer sees the Batik classes. Something like that. It gives me headache 😂   

It seems to work if I provide some additional libraries like Apache Batik and Xerces. I continue my tests.

**Comment by jonathanlermitage** [comment]
@jmatias can you please try this build 
[Extra Icons-2023.1.1.zip](https://github.com/jonathanlermitage/intellij-extra-icons-plugin/files/10844486/Extra.Icons-2023.1.1.zip)
? It works for me, on IJ EAP and IJ Stable. I tried to add a custom icon, and edit a custom icon.

**Comment by jmatias** [comment]
Will do

**Comment by jmatias** [comment]
Still getting the same error. I now see this in the idea.log




 WARN - #c.i.u.s.SvgTranscoder - data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD//gAfQ29tcHJlc3NlZCBieSBqcGVnLXJlY29tcHJlc3P/2wCEAAQEBAQEBAQEBAQGBgUGBggHBwcHCAwJCQkJCQwTDA4MDA4MExEUEA8QFBEeFxUVFx4iHRsdIiolJSo0FVQtFKtc8j6P ....  url.uninterpretable

**Comment by jonathanlermitage** [comment]
This SVG file contains a JPEG image. Unfortunately, this no longer works with IJ 2023 EAP, and I found no solution.  
As a workaround, you may want to use a JPEG file instead.

**Comment by jonathanlermitage** [comment]
I reopen this issue. This fix breaks IJ 2022.3.2, and I found some scenarios where IJ 2023 is broken too.

**Comment by jonathanlermitage** [comment]
Found a way to render SVG files without the standard ImageIO plugin system. I directly use Batik. Thx https://stackoverflow.com/questions/11435671/how-to-get-a-bufferedimage-from-a-svg 😅
